### PR TITLE
fix(storage): automatically configure crc32c for ARM64

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12851)

### CODE REVIEW

File: bazel/crc32c.BUILD [hunk 1]
```diff
@@ -49,70 +52,79 @@ crc32c_SRCS = [
...
-})
-
-load(":configure_template.bzl", "configure_template")
-
 configure_template(
     name = "generate_config",
     output = "crc32c/crc32c_config.h",
     substitutions = {
         "#cmakedefine01": "#define",
+        # On x86_64 endianness is always little end.  ARM64 is bi-endian,
```
Comment (devbww) [inline bazel/crc32c.BUILD:93]: s/end/endian/?

Or just "x86_64 is always little endian."

File: bazel/crc32c.BUILD [hunk 2]
```diff
@@ -49,70 +52,79 @@ crc32c_SRCS = [
...
+        # fortunately.  macOS and Windows only use little endian.
+        #    https://devblogs.microsoft.com/oldnewthing/20220726-00
+        #    https://developer.apple.com/documentation/apple-silicon/porting-your-macos-apps-to-apple-silicon
+        # We are going to assume the same is true for Linux.
```
Comment (devbww) [inline bazel/crc32c.BUILD:97]: s/are going to //

File: bazel/crc32c.BUILD [hunk 3]
```diff
@@ -49,70 +52,79 @@ crc32c_SRCS = [
...
         " HAVE_BUILTIN_PREFETCH": " HAVE_BUILTIN_PREFETCH 0",
         " HAVE_MM_PREFETCH": " HAVE_MM_PREFETCH 0",
-        " HAVE_SSE42": " HAVE_SSE42 1",
-        " HAVE_ARM64_CRC32C": " HAVE_ARM64_CRC32C 0",
         " HAVE_STRONG_GETAUXVAL": " HAVE_STRONG_GETAUXVAL 0",
         " HAVE_WEAK_GETAUXVAL": " HAVE_WEAK_GETAUXVAL 0",
         " CRC32C_TESTS_BUILT_WITH_GLOG": " CRC32C_TESTS_BUILT_WITH_GLOG 0",
-    },
+    } | select({
+        # We are going to assume all x86_64 CPUs support SSE4.2. This seems
```
Comment (devbww) [inline bazel/crc32c.BUILD:105]: s/are going to //

File: bazel/crc32c.BUILD [hunk 4]
```diff
@@ -49,70 +52,79 @@ crc32c_SRCS = [
...
+        # reasonably safe in 2023.
+        "@platforms//cpu:x86_64": {
+            " HAVE_SSE42": " HAVE_SSE42 1",
+            " HAVE_ARM64_CRC32C": " HAVE_ARM64_CRC32C 0",
+        },
+        # We are going to assume all ARM64 CPUs support CRC32C extensions. This
```
Comment (devbww) [inline bazel/crc32c.BUILD:111]: s/are going to //

**Review by devbww — APPROVED** [review]

File: bazel/crc32c.BUILD (hunk 1 above)
Comment (coryan) [reply bazel/crc32c.BUILD]: Done.

File: bazel/crc32c.BUILD (hunk 2 above)
Comment (coryan) [reply bazel/crc32c.BUILD]: Done.

File: bazel/crc32c.BUILD (hunk 3 above)
Comment (coryan) [reply bazel/crc32c.BUILD]: Done.

File: bazel/crc32c.BUILD (hunk 4 above)
Comment (coryan) [reply bazel/crc32c.BUILD]: Done.

**Pull request merged**